### PR TITLE
Don't process HEP messages before sending out

### LINKED ISSUE #2251: [BUG]
**OpenSIPS version you are running**

```
version: opensips 3.1.0 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 58804282f
main.c compiled on  with gcc 8

```

**Describe the bug**
When using both tracer and topology_hiding on 3.1, logs are filled with lots of 
`ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?`
messages.

It looks like if topology_hiding was trying to modify hep packets before they get sent (probably to hide topology informations) but is unable to parse HEP packets (what seems quite normal).

**To Reproduce**
Using trace functions and topology hiding on the same script

**Expected behavior**
Not having logs filled with these messages.
Probably by having a way to tell topology_hiding to ignore locally generated and sent HEP packets.

**Relevant System Logs**
Sep 17 13:02:34 [26446] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:34 [26446] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:34 [26438] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:37 [26444] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:37 [26434] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:37 [26434] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3
Sep 17 13:02:37 [26434] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:37 [26434] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3?
Sep 17 13:02:37 [26447] ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3N

### LINKED ISSUE #2273: [BUG] topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3
**OpenSIPS version you are running**
```
3.1.0
git revision: 3319120fb
```

**Describe the bug**
I see that it was described here https://github.com/OpenSIPS/opensips/commit/7ada147dd074b6bd058e5e0e67fac6ad55613f27
I have a lot of this messages in my log, but don't have any idea why it happens. Google point out only on this commit )
In my case it happens when topology_hidding() was called (with create dialog a/b) for initial requests and trace() called after that with capturing transactions and dialogs ("td"). When I just trace messages ("m") it does not show this errors.
Maybe author @lemenkov could provide some hint/idea what it can be?
Ps. Of course I understand that it's homer proto v3, but why it incomes to topology_hiding logic?

**To Reproduce**
```
  # inside initial requests logic
  if (!$DLG_status) {
    if (!create_dialog() || !topology_hiding("C")) {
      xlog("L_ERR","[TH] $rm - $C(rx)can NOT create hide topology!$C(xx)\n");  
      exit;
    }    
    # we can trace dialog only after creation, otherwise the dialog has not be created.
    trace("hid", "mtd", "sip|xlog");
   ...
```

**Relevant System Logs**
```
ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: HEP3
```

**OS/environment information**
 - Operating System: Centos 7
 - OpenSIPS installation: nightly rpm

### PR DESCRIPTION
**Pull request by lemenkov**
These mesages are not intended for any further processing so don't even
try changing anything.

Fixes #2251, fixes #2273.

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

### CODE REVIEW

**Comment by lemenkov** [comment]
Still waiting for the comments (commenting just to remove "stale" tag).

**Comment by volga629-1** [comment]
The same message is appear with  PUBLISH  on  INVITE reply.

```
Jan  1 08:04:18 voice-proxy /usr/sbin/opensips[942581]: OnReply_Route3: [INVITE]  Direction: [FS ~> Client] and source IP 10.30.0.8 and media in-iface=media out-iface=outside with transport RTP/AVP
Jan  1 08:04:18 voice-proxy /usr/sbin/opensips[942581]: ERROR:topology_hiding:topo_callid_post_raw: could not parse resulted sip message: PUBLISH

```

File: forward.h [hunk 1]
```diff
@@ -114,7 +114,7 @@ static inline int msg_send( struct socket_info* send_sock, int proto,
 	/* the raw processing callbacks are free to change whatever inside
 	 * the buffer further use out_buff.s and at the end try to free out_buff.s
 	 * if changed by callbacks */
-	if (proto != PROTO_BIN)
+	if ((proto != PROTO_BIN) && (proto != PROTO_HEP_TCP) && (proto != PROTO_HEP_UDP))
```
Comment (bogdan-iancu) [inline forward.h:117]: @lemenkov , just to make things clear here, maybe we should add in `ip_addr.h` some new macro to tell if a protocol is SIP based or not - this is actually what we need here, to rule out the none-SIP protos.
Maybe something like `PROTO_IS_SIP()` ?

Comment (lemenkov) [reply forward.h:117]: @bogdan-iancu yes  this would be better!

Comment (bogdan-iancu) [reply forward.h:117]: Could you check the eee990d commit (master) as an alternative to your patch? Upon your confirmation and I will do the backport. Thanks @lemenkov

Comment (lemenkov) [reply forward.h:117]: @bogdan-iancu looks good to me (rolled into production for a while now)

**Comment by bogdan-iancu** [comment]
OK, backports to 3.1 and 2.4 done , thank you @lemenkov

**Pull request closed without merging**